src/modules/todo.js

```diff
@@ -0,0 +1,50 @@
+import { storeData, retrieveData } from './handleData.js';
+
+let newTodoArray = [];
+
+export default class Todo {
+  constructor(description, completed, index) {
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+
+  static getTodo = (index) => {
+    newTodoArray = retrieveData();
+    const todo = newTodoArray.find((x) => x.index.toString() === index.toString());
+    return todo;
+  }
+
+  static addTodo = (todo) => {
+    const newTodo = new Todo(
+      todo.description,
+      todo.completed,
+      todo.index,
+    );
+    newTodoArray = retrieveData();
+    newTodoArray.push(newTodo);
+    storeData(newTodoArray);
+  }
+
+  static updateTodo = (todo) => {
+    newTodoArray = retrieveData();
+    newTodoArray = newTodoArray.filter((element) => element.index !== todo.index);
+    const newTodo = new Todo(
+      todo.description,
+      todo.completed,
+      todo.index,
+    );
+    newTodoArray.push(newTodo);
+    storeData(newTodoArray);
+  }
+
+  static removeTodo = (index) => {
+    newTodoArray = retrieveData();
+    newTodoArray = newTodoArray.filter((element) => element.index.toString() !== index.toString());
+    const reIndexedArray = [];
+    newTodoArray.forEach((element, index) => {
+      reIndexedArray.push(new Todo(element.description, element.completed, index + 1));
+    });
+    storeData(reIndexedArray);
+  }
+}
```

Comment:
- [ ] As required for the project, kindly make sure your application is able to implement the following functionalities:
  - A functionality for editing task descriptions should be implemented.
  - A functionality for deleting a task (remove an element from the array) should be implemented.
  - Once you have implemented the deleting functionality a task should, all remaining items' indexes, so they represent the current list order and are unique.
  - The important thing, you should make sure all changes to the To-Do List are correctly saved in local storage(`editing`, `deleting` a task,...)

Below, you can see how your application behavior as it can only `add` a task but not `deleting` and  `editing`:

![todo](https://user-images.githubusercontent.com/83042742/213079087-1d0bfbc1-1eed-4566-b792-972686baa7dc.gif)